### PR TITLE
:bug: Pinned-Dependencies: Fix invalid shell file handling

### PR DESCRIPTION
**Pull request by lehors**
Parsing errors are meant to be discarded but aren't. This patch
changes the code so that the error is indeed discarded and checking
continues as intended.

Signed-off-by: Arnaud J Le Hors <lehors@us.ibm.com>

* **Please check if the PR fulfills these requirements**
- [ x] Tests for the changes have been added (for bug fixes / features)
- [ x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
See Issue #1307 

* **What is the new behavior (if this is a feature change)?**
The parsing error is discarded as intended and a score is given as a result.
The bottom of the description of Issue #1307 shows the new output.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

### CODE REVIEW

File: checks/shell_download_validate.go [hunk 1]
```diff
@@ -914,6 +914,7 @@ func validateShellFile(pathfn string, content []byte, dl checker.DetailLogger) (
 	if err != nil && errors.Is(err, sce.ErrorShellParsing) {
 		// Discard and print this particular error for now.
 		dl.Debug(err.Error())
+		err = nil
```
Comment (azeemshaikh38) [inline checks/shell_download_validate.go:917]: Explicit `return false, nil` here since value of `r` is undefined.

**Review by azeemshaikh38 — APPROVED** [review]
Thanks @lehors !

File: checks/shell_download_validate.go (hunk 1 above)
Comment (lehors) [reply checks/shell_download_validate.go:917]: `r` is not undefined actually. It is set to false. See: https://github.com/ossf/scorecard/blob/23b0ddb8aa96356321cf31a2709723e29b15a951/checks/shell_download_validate.go#L771

File: checks/shell_download_validate_test.go [hunk 2]
```diff
@@ -86,3 +88,18 @@ func TestIsSupportedShellScriptFile(t *testing.T) {
 		})
 	}
 }
+func TestValidateShellFile(t *testing.T) {
```
Comment (chrismcgehee) [inline checks/shell_download_validate_test.go:91]: If you just add a newline before this method and add
```
	t.Parallel()
```
at the beginning of this method, the linter should pass.

Other than that, looks great! Thanks for fixing this!

**Review by chrismcgehee — CHANGES_REQUESTED** [review]

File: checks/shell_download_validate_test.go (hunk 2 above)
Comment (lehors) [reply checks/shell_download_validate_test.go:92]: My bad. I fixed it. Thanks.

**Review by chrismcgehee — APPROVED** [review]

**Pull request merged**